### ISSUE TITLE
Create new folder named -p on windows

### ISSUE THREAD

**Issue by ERROPiX** [ahmadawais/create-guten-block#60]
Hi,

I just tried this plugin, but sometimes it fails, sometimes it fails with errors, and it create two folders, one with name I given, and other nammed **-p**

Thanks

**Comment by ahmadawais** [comment]
That's strange. @mrsaqibameen can you test this. I don't own a Windows system.

**Comment by mrsaqibameen** [comment]
@ahmadawais I did some tests, here are the results:

- Yes, it creates a folder named **-p**, whenever I create a new block.
- I have created three blocks, everytime it created block successfully. No failure.
- After creating first block, whenever I created another block, it gave error `A subdirectory or file -p already exists. Error occurred while processing: -p.` but successfully created the block.

Basically it is creating an extra folder named **-p** whenever I create a new block. Rest it's fine.

**Comment by Varstahl** [comment]
I can confirm the issue, I used the npx command for the first time and along the block directory there was a `-p` empty folder.

**Comment by Varstahl** [comment]
I'm sorry, I should have explained myself better. What I meant is that I got a fresh installation of the latest Node.JS, and as per your readme I used the command `npx create-guten-block <blockname>`. If there are other ways of doing the same without npx, I didn't try them. Anyway here's the output of the envinfo:

```
> npx envinfo
npx: installed 1 in 2.889s
The "path" argument must be of type string. Received type undefined
npx: installed 1 in 2.925s
%AppData%\npm-cache\_npx\9232\node_modules\envinfo\dist\cli.js

  System:
    OS: Windows 7
    CPU: x64 Intel(R) Core(TM) i7-4930K CPU @ 3.40GHz
    Memory: 36.09 GB / 48.00 GB
  Binaries:
    Node: 10.2.1 - …\node-v10.2.1-win-x64\node.EXE
    npm: 5.6.0 - …\node-v10.2.1-win-x64\npm.CMD
  Languages:
    Python: 3.6.0 - …\Python36\python.EXE
  Browsers:
    Internet Explorer: 11.0.9600.19003
```

**Comment by Varstahl** [comment]
Sure thing:

![-p directory](https://user-images.githubusercontent.com/284077/40586814-a55dd852-61c7-11e8-9c9c-4dc8f83021ef.png)

During step 1 and before installing the npm packages, 2 directories are created instead of one. ~From a logic standpoing I'm assuming the script runs something like `mkdir -p <scriptname>`, which works fine on osx and *nix, but since there is no such option in Win it treats the non-existent option as two separate directories, creating both `-p` and `<scriptname>`.~ Exactly what happens.

[Well, found the problem](/ahmadawais/create-guten-block/blob/master/packages/create-guten-block/app/createPluginDir.js#L45).

**Comment by Varstahl** [comment]
@ahmadawais Windows automatically creates parents if they're not present, so you must suppress the -p for the platform.

**Comment by ahmadawais** [comment]
Here's the line that causes the issue. I am using ShellJS and it's the issue with that library.

https://github.com/ahmadawais/create-guten-block/blob/master/packages/create-guten-block/app/createPluginDir.js#L45

**Comment by Varstahl** [comment]
Yup, that's the line I spotted. Glad I could help :)
